compiler/src/dmd/expression.d

```diff
@@ -2373,7 +2372,7 @@ extern (C++) final class StructLiteralExp : Expression
     // while `sym` is only used in `e2ir/s2ir/tocsym` which comes after
     union
     {
-        Symbol* sym;            /// back end symbol to initialize with literal
```

Comment:
I think this is incomplete. There's also typedefs to deal with.

Regardless, I don't think this is as small a change for C++ codebases as it is for D. So hold fire.

Reply:
I thought `Symbol` was dmd-only, so it wouldn't affect other compilers?

Reply:
Symbol seems to be an alias to a tree_node in gcc. I am expecting that all you need to do on the GCC side is to replace all occurences of `sym` to `(tree_node*)sym`.

Reply:
@dkorpel the gluelayer has a version(IN_GCC) that alias Symbol to an extern gcc symbol (tree_node). Still, I don't see how this makes things complicated for gdc.

Reply:
> Regardless, I don't think this is as small a change for C++ codebases as it is for D. So hold fire.

I highly doubt it. At most it requires the insertion of a couple of pointer casts. 

LDC `#if`s out the use of `Symbol` in `module.h` so the typedefs are a non-issue for us. Indeed `dmd.gluelayer` does not define `Symbol` with `version (IN_LLVM)`, at all. The usage here is `void*` in D and `llvm::Type*` in the header file.

It has all of 4 uses which would add 4 pointer casts.

Also this PR affects only `StructLiteralExp` it doesn't touch any other uses of `Symbol`